### PR TITLE
Extends contributing documentation, adds CSV parser type hints...  #69

### PR DESCRIPTION
**Pull request by mohammad-suliman**
supersedes #69

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,26 +1,165 @@
...
+
+
+```Bash
+npm run py-server
+```
+
+
+**Preview Changes**
+
+visit "http://localhost 8080"
```
Comment (mohammad-suliman) [inline CONTRIBUTING.md:84]: @oferb, can you please verify that this looks as you expect?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,26 +1,165 @@
...
+
+> Note, if Python version 3 is not available, it is possible, at this time, to instead preview via visiting:
+file:///<path_to_repo>/public/index.html
+
+> ... however, this is **not** recommended as some web-browsers do not take kindly to loading JavaScript from `file:///` paths, and some web-browsers really do not like loading JavaScript this way when there are remote JavaScript files involved, eg. JQuery.
```
Comment (mohammad-suliman) [inline CONTRIBUTING.md:89]: @oferb, And those lines also please...

Comment (oferb) [reply CONTRIBUTING.md:89]: Yup

File: CONTRIBUTING.md (hunk 1 above)
Comment (oferb) [reply CONTRIBUTING.md:84]: It's good

**Pull request merged**